src/app/services/tasksApi.ts

```diff
@@ -1,14 +1,27 @@
-import task from '@/interfaces/task.type';
+import task, { updateTaskDetails } from '@/interfaces/task.type';
 import { api } from './api';
-import { MINE_TASKS_URL } from '@/constants/url';
+import { MINE_TASKS_URL, TASKS_URL } from '@/constants/url';
+TASKS_URL;
 
 type TasksQueryResponse = { message: string; tasks: task[] };
+type TasksCreateMutationResponse = { message: string; task: task };
+type TaskRequestPayload = { task: updateTaskDetails; id: string };
 
 export const tasksApi = api.injectEndpoints({
     endpoints: (builder) => ({
         getAllTasks: builder.query<TasksQueryResponse['tasks'], void>({
             query: () => '/tasks',
-            providesTags: ['Tasks'],
+            providesTags: (result) =>
```

Comment:
want to know....what result parameter contain??
type of result parameter?? :)

Reply:
`TasksQueryResponse['tasks']` this would be the type